### PR TITLE
docs(blog): add bigquery arrays 7.0.0 blog post

### PR DESCRIPTION
**Pull request by cpcloud**
This PR adds a blog post showcasing array API usage with the BigQuery backend whose support for array APIs was expanded in 7.0.0.

### CODE REVIEW

File: docs/posts/bigquery-arrays/index.qmd [hunk 1]
```diff
@@ -0,0 +1,231 @@
+# Working with arrays in Google BigQuery
```
Comment (lostmygithubaccount) [inline docs/posts/bigquery-arrays/index.qmd:1]: can you add the frontmatter? example from v6.1.0 release blog:

```
---
title: Ibis v6.1.0
author: "Ibis team"
date: "2023-08-02"
categories:
    - release
    - blog
---
```

Comment (cpcloud) [reply docs/posts/bigquery-arrays/index.qmd:1]: Yep, thanks!

File: docs/posts/bigquery-arrays/index.qmd [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+We can see the position of `"actor"` in `primary_profession`s:
+
+```{python}
+ents.primary_profession.index("actor")
+```
+
+A return value of `-1` indicates that `"actor"` is not present in the value:
+
+Let's check whether `"actor"` shows up in a different position in the `primary_profession` column:
```
Comment (gforsyth) [inline docs/posts/bigquery-arrays/index.qmd:100]: This sentence threw me for a second (although I'm not looking at the rendered output, either).

Maybe a touch more explicit prose re: looking for entities who aren't primarily actors?

File: docs/posts/bigquery-arrays/index.qmd [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+```
+
+## Advanced operations
+
+### `unnest`
+
+As of version 7.0.0 Ibis does not support its native `unnest` API for BigQuery,
+but we plan to add it in the future.
+
+For now, you can use `con.sql` to construct an ibis expression from a BigQuery
```
Comment (gforsyth) [inline docs/posts/bigquery-arrays/index.qmd:198]:
```suggestion
For now, you can use `con.sql` to construct an Ibis expression from a BigQuery
```

**Review by gforsyth — COMMENTED** [review]
looking good!  a few small missing words and tweaks

File: docs/posts/bigquery-arrays/index.qmd (hunk 2 above)
Comment (cpcloud) [reply docs/posts/bigquery-arrays/index.qmd]: Reworded

File: docs/posts/bigquery-arrays/index.qmd [hunk 4]
```diff
@@ -0,0 +1,247 @@
+---
+title: Working with arrays in Google BigQuery
+author: "Phillip Cloud"
+date: "2023-09-12"
+categories:
+    - release
```
Comment (lostmygithubaccount) [inline docs/posts/bigquery-arrays/index.qmd:6]: are you intending this as a "release blog?" -- I was thinking of the `release` category as a running series of blogs that cover any release with major-ish features more indepth than release notes

File: docs/posts/bigquery-arrays/index.qmd [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+---
+
+## Introduction
+
+Ibis and BigQuery have [worked well together for years](https://cloud.google.com/blog/products/data-analytics/ibis-and-bigquery-scalable-analytics-comfort-python).
+
+In Ibis 7.0.0, they work even better together with the addition of array
+functionality for BigQuery.
+
+Let's look at some examples using BigQuery's [IMDB data](https://developer.imdb.com/non-commercial-datasets/).
```
Comment (lostmygithubaccount) [inline docs/posts/bigquery-arrays/index.qmd:20]: nit: perhaps "BigQuery's sample IMDB data"?

File: docs/posts/bigquery-arrays/index.qmd [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+```
+
+It's not 100% clear whether the order of elements in `primary_profession` matters here.
+
+### Containment
+
+We can get people who are **not** actors using `contains`:
+
+```{python}
+non_actors = ents[~_.primary_profession.contains("actor")]
```
Comment (lostmygithubaccount) [inline docs/posts/bigquery-arrays/index.qmd:130]: this line could use a quick comment/annotation/explanation somewhere IMO with the `~_.`, I think that could be confusing for a newer reader

Comment (lostmygithubaccount) [reply docs/posts/bigquery-arrays/index.qmd:130]: perhaps reference the selectors blog/documentation somewhere

File: docs/posts/bigquery-arrays/index.qmd [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+```
+
+## Conclusion
+
+Ibis has a sizable collection of array APIs that work with many different
+backends and as of version 7.0.0, Ibis supports a much larger set of those APIs
+for BigQuery!
+
+Check out [the API
+documentation](http://localhost:8000/reference/expression-collections.html#ibis.expr.types.arrays.ArrayValue)
```
Comment (lostmygithubaccount) [inline docs/posts/bigquery-arrays/index.qmd:244]: looks like this link needs an update

**Review by lostmygithubaccount — APPROVED** [review]
LGTM, minor suggestions (one link fix I think)

File: docs/posts/bigquery-arrays/index.qmd (hunk 4 above)
Comment (cpcloud) [reply docs/posts/bigquery-arrays/index.qmd:6]: Nope, thanks!

File: docs/posts/bigquery-arrays/index.qmd (hunk 7 above)
Comment (cpcloud) [reply docs/posts/bigquery-arrays/index.qmd:244]: Yep, forgot to push this fix.

File: docs/posts/bigquery-arrays/index.qmd (hunk 6 above)
Comment (cpcloud) [reply docs/posts/bigquery-arrays/index.qmd:130]: It's not a selector that I'm using here, this is a filter expression.

File: docs/posts/bigquery-arrays/index.qmd (hunk 5 above)
Comment (cpcloud) [reply docs/posts/bigquery-arrays/index.qmd]: Fixed.

File: docs/posts/bigquery-arrays/index.qmd (hunk 7 above)
Comment (cpcloud) [reply docs/posts/bigquery-arrays/index.qmd]: Pushed!

**Pull request merged**